draft-ietf-rats-corim.md

```diff
@@ -1429,9 +1429,173 @@ is extended using Endorsements etc. from the accepted tags.
+value with handling described in the sub-sections below,
+then the algorithm appropriate to that key is used to match the entries.
+
+If the Reference Value is not tagged, and the `measurement-values-map` key
+is not a value described below, then the entries are compared
+using binary comparison of their CBOR encoded values. If the values
+are not binary identical then the Reference Value does not match.
+
+Note that while specifications may extend the matching semantics using CBOR tags,
+there is no way to extend the matching semantics of key values.
+Any new keys requiring non-default comparison must add a CBOR tag to the
+reference value describing the desired behaviour.
+
+If the Reference Value contains an `authorized-by` field then the Verifier
+SHALL check for an `authorized-by` field in the Accepted Claims Set entry
+compared in the steps above. If the Accepted Claims Set key is not one of
+the keys from the reference value `authorized-by` field then the
+reference value does not match.
+
+If all checks above have been performed successfully then the Reference Value
+matches.
+
+##### Comparison for svn entries
+
+The value stored under `measurement-values-map` key 1 is an SVN, which must
+have type UINT.
+
+If the Reference value for `measurement-values-map` key 1 is an untagged UINT or
+a UINT tagged with #6.552 then an equality comparison is performed. If the value
+of the SVN in Accepted Claims Set is not equal to the value in the Reference
+Value then the Reference Value does not match.
+
+If the Reference value for `measurement-values-map` key 1 is a UINT tagged with
+#6.553 then a minimum comparison is performed. If the value of the SVN in
+Accepted Claims Set less than the value in the Reference Value then the
+Reference Value does not match.
+
+##### Comparison for digests entries
+
+The value stored under `measurement-values-map` key 2, or a value tagged with
+#6.TBD is a digest entry. It contains one or more digests, each measuring the
+same object. A Reference Value may contain multiple digests, each with a
+different algorithm acceptable to the Reference Value provider. If the
+digest in evidence contains a single value with an algorithm and value
+matching one of the algorithms and values in the Reference Value then it
+matches.
+
+To prevent downgrade attacks, if there are multiple algorithms which are in
+both the evidence and Reference Value then the digests calculated using all
+shared algorithms must match.
+
+If the CBOR structure of the digest entry in the Reference Value or the
+Accepted Claim Set value with the same key is incorrect then the Reference
+Value does not match.
+
+The Verifier MUST iterate over the reference value `digests` array, locating
+algorithms which are present in the reference value and the Accepted Claims
+Set entry.
+
+If there is no hash algorithm identifier which is present in the Reference Value
+and the Accepted Claims Set entry then the Reference Value does not match.
```

Comment:
```suggestion
differs from the hash algorithm identifier in the Accepted Claims Set entry then the Reference Value does not match.
```

Reply:
Note: IANA registry for hash algs allows multiple encodings: numeric (0,1,2,...), stringified numeric ("0", "1", "2",...) and strings "sha-256", "sha-384" etc. Should the description here point to the IANA registry and should the Verifier perform reasonable similarity checking, e.g.: "1" equals 1 equals "sha-256"?

Additionally, profiles can define their own alg ids (so long as they don't conflict with the IANA registry).

Reply:
yes on the suggestion (as above). @nedmsmith, maybe rescue your comment as an issue?